lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,428 @@
+
+    # Note: user id can also be retrieved from the encoded auth header
+    @auth.login_required
+    def get(self, username):
+        """
+        Get request for getting user data.
+        This requires a Pbench auth token in the header field
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            Authorization:  Bearer Pbench_auth_token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "message": "Successfully registered."/"failure message",
+                    "data": {
+                        "username": <username>,
+                        "firstName": <firstName>,
+                        "lastName": <lastName>,
+                        "registered_on": registered_on,
+                    }
+                }
+        """
+        try:
+            user, verified = verify_user(username)
+        except Exception:
+            self.logger.exception("Exception occurred during the getUser")
+            abort(500, message="INTERNAL ERROR")
+
+        # TODO: Check if the user has the right privileges
+        if verified or user.is_admin():
+            response_object = {
+                "status": "success",
+                "data": {
+                    "email": user.email,
+                    "firstName": user.firstName,
+                    "lastName": user.lastName,
+                    "registered_on": user.registered_on,
+                },
+            }
+            return make_response(jsonify(response_object), 200)
+        else:
+            self.logger.warning(
+                "Username retrieved from the auth token {} is different from the username provided",
+                auth.current_user().id,
+            )
+            abort(403, message="Authentication token does not belong to the user")
+
+    @auth.login_required
+    def delete(self, username):
+        """
+        Delete request for deleting a user from database.
+        This requires a Pbench auth token in the header field
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            Authorization:   Bearer Pbench_auth_token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 200/status_code_int,
+                    "message": "Successfully deleted."/"failure message",
+                }
+        """
+        try:
+            user, verified = verify_user(username)
+        except Exception:
+            self.logger.exception("Exception occurred during the getUser")
+            abort(500, message="INTERNAL ERROR")
+
+        # TODO: Check if the user has the right privileges
+        if not verified and not user.is_admin():
```

Comment:
So for this boolean logic, we are taking action to say that if a user is not verified and is not an admin, the "Authentication token does not belong to the user".  But what does it mean for "not verified, but is an Admin..."?  Why would we allow a user who is not verified (`username` given does not match the `Authorization` header) to proceed when it is an admin user?

Reply:
We would proceed if the auth belongs to an admin user because if an admin wants to delete a user from the database, he should be able to do that. So an admin user with his own auth token can mention some other username in the URI that he wants to delete (in this case the username is not verified because the auth token does not belong to that user) but since the user performing the action is admin we would proceed to delete the user with that username.

Reply:
Just had a thought here: if we allow the "admin" user to delete the "admin" user, we've got a problem. At best, we need a mechanism to restore it... but maybe we shouldn't allow that in the first place?

Reply:
yes I have added `is_admin` check before deleting the user